sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -265,78 +257,53 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 		}
 	}
 
-	// In Kubernetes, the `default` service account always exists in each namespace (it is recreated upon deletion).
-	// The default service account always contains an API token.
-	// In OpenShift, the default service account also contains credentials for the
-	// OpenShift Container Registry, which is an internal image registry.
-	fromDefaultSA := saName == defaultSA
-	isGlobalPullSecret := secret.GetNamespace() == openshiftConfigNamespace && secret.GetName() == openshiftConfigPullSecret
+	s.processSecretForLocalScanning(secret, action, dockerConfig, saName)
 
 	newIntegrationSet := set.NewStringSet()
 	for registryAddress, dce := range dockerConfig {
 		registryAddr := strings.TrimSpace(registryAddress)
 		if registryAddr != registryAddress {
-			log.Warnf("Spaces have been trimmed from registry adress %q found in secret %s/%s",
+			log.Warnf("Spaces have been trimmed from registry address %q found in secret %s/%s",
 				registryAddress, secret.GetNamespace(), secret.GetName())
 		}
-		if fromDefaultSA {
-			// Store the registry credentials so Sensor can reach it.
-			err := s.regStore.UpsertRegistry(context.Background(), secret.GetNamespace(), registryAddr, dce)
-			if err != nil {
-				log.Errorf("Unable to upsert registry %q into store: %v", registryAddr, err)
-			}
 
-			s.regStore.AddClusterLocalRegistryHost(registryAddr)
+		registries = append(registries, &storage.ImagePullSecret_Registry{
+			Name:     registryAddr,
+			Username: dce.Username,
+		})
 
-		} else if saName == "" {
-			// only send integrations to central that do not have the k8s SA annotation
-			// this will ignore secrets associated with OCP builder, deployer, etc. service accounts
-			ii, err := DockerConfigToImageIntegration(secret, registryAddr, dce)
-			if err != nil {
-				log.Errorf("unable to create docker config for secret %s: %v", secret.GetName(), err)
-			} else if !managedcentral.IsCentralManaged() {
-				sensorEvents = append(sensorEvents, &central.SensorEvent{
-					// Only update is supported at this time.
-					Action: action,
-					Resource: &central.SensorEvent_ImageIntegration{
-						ImageIntegration: ii,
-					},
-				})
-				if features.SourcedAutogeneratedIntegrations.Enabled() {
-					newIntegrationSet.Add(ii.GetId())
-				}
-			}
+		// Only send integrations to Central that are not bound to a service account and managed by k8s.
+		// This will ignore the secrets generated for the OCP default, builder, deployer, etc. service accounts.
+		// These pull secrets are used only for accessing the OCP internal registry, which is only accessible
+		// from within the Secured Cluster. Central will be unable to use these credentials.
+		if saName != "" {
```

Comment:
Can you rename this plus `saAnnotations`? I keep thinking these are generic, and not OCP-specific. Can you make them something that makes it clear they are OCP-specific?

Reply:
'Technically' they are NOT all OCP specific - we are using them for an OCP use case but technically the annotations can appear in non-openshift scenarios. Such as [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#:~:text=a%20special%20annotation%2C-,kubernetes.io/service%2Daccount.name,-.).